public/app/features/dashboard/state/PanelQueryRunner.ts

```diff
@@ -158,7 +157,20 @@ export class PanelQueryRunner {
       request.interval = norm.interval;
       request.intervalMs = norm.intervalMs;
 
-      this.pipeToSubject(runRequest(ds, request));
+      const reqObservable = runRequest(ds, request);
+      this.pipeToSubject(
+        reqObservable.pipe(
+          throttle(
+            ev => {
+              return merge(
+                interval(1000),
+                reqObservable.pipe(filter((data: PanelData) => data.state === LoadingState.Done))
```

Comment:
this block of codes needs some comments :)  how can you use the same observable twice without share()?

Reply:
Definitely still need some polish and add comments but wanted another opinion on the issue with the combined state.

Here it works because `runRequest` already returns observable that is using `share()` but yeah that is totally not obvious here (and no special type for that either). Maybe using `share()` again here would make sense so we do not rely on internal implementation of `runRequest`.

Reply:
Ok actually this does not seem to work right now so need to investigate :(